File: 2025/go/day02/part1.go
package main

import (
	"errors"
	"fmt"
	"os"
	"strconv"
	"strings"
)

const FileName = "input.txt"

func ReadLines(fileName string) ([]string, error) {
	data, err := os.ReadFile(fileName)

	if err != nil {
		return nil, errors.New("we cannot open file")
	}

	contents := string(data)

	contents = strings.ReplaceAll(contents, "\n", "")
	contents = strings.ReplaceAll(contents, "\r", "")

	parts := strings.Split(contents, ",")

	var result []string

	for _, p := range parts {
		p = strings.TrimSpace(p)

		if p != "" {
			result = append(result, p)
		}
	}

	return result, nil
}

func part1() {
	lines, err := ReadLines(FileName)

	if err != nil {
		panic(err)
	}

	var invalidIds []int
	for i, _ := range lines {
		lId, err := strconv.Atoi(strings.Split(lines[i], "-")[0])
		if err != nil {
			panic("We cannot parse Left Index")
		}
		rId, err := strconv.Atoi(strings.Split(lines[i], "-")[1])

		if err != nil {
			panic("We cannot parse Right Index")
		}

		prevNum := lId

		for j := lId; j < rId; j++ {
			if strconv.Itoa(prevNum)[0:len(strings.Split(strconv.Itoa(prevNum), "")) / 2] ==
			strconv.Itoa(prevNum)[len(strings.Split(strconv.Itoa(prevNum), "")) / 2:] {
				invalidIds = append(invalidIds, prevNum)
			}

			prevNum = prevNum + 1
		}

		if strconv.Itoa(rId)[0:len(strings.Split(strconv.Itoa(rId), "")) / 2] ==
		strconv.Itoa(rId)[len(strings.Split(strconv.Itoa(rId), "")) / 2:] {
			invalidIds = append(invalidIds, rId)
		}
	}

	var sum int
	for _, value := range invalidIds {
		sum += value
	}

	fmt.Println(sum)
}

File: 2025/js/day02/part1.mjs
import fs from 'fs';

const fileName = "part1.txt";

const contents = fs.readFileSync(fileName, {
    encoding: "utf8"
});


const contentsNumbers = contents.replaceAll("\n", "").split(",");
const invalidIds = [];

for(let i = 0; i < contentsNumbers.length; i++) {
    const lId = +contentsNumbers[i].split("-")[0]
    const rId = +contentsNumbers[i].split("-")[1]

    let prevNum = lId;

    for(let j = lId; j < rId; j++) {
        if(prevNum.toString().substring(0, prevNum.toString().split("").length / 2) ===
            prevNum.toString().substring(prevNum.toString().split("").length / 2)
        ) {
            invalidIds.push(prevNum);
        }


        prevNum = prevNum + 1;
    }

    if(rId.toString().substring(0, rId.toString().split("").length / 2) ===
        rId.toString().substring(rId.toString().split("").length / 2)
    ) {
        invalidIds.push(rId);
    }
}

const sum = invalidIds.reduce((acc, cum) => acc + cum, 0)

console.log(sum)
